addons/wiki/models.py

```diff
@@ -182,8 +249,42 @@ def absolute_api_v2_url(self):
     def get_absolute_url(self):
         return self.absolute_api_v2_url
 
+class WikiPageNodeManager(models.Manager):
+
+    def create_for_node(self, node, name, content, auth):
+        existing_wiki_page = WikiPage.objects.get_for_node(node, name)
+        if existing_wiki_page:
+            raise NodeStateError('Wiki Page already exists.')
+
+        wiki_page = WikiPage.objects.create(
+            node=node,
+            page_name=name,
+            user=auth.user
+        )
+        # Creates a WikiVersion object
+        wiki_page.update(auth.user, content)
+        return wiki_page
+
+    def get_for_node(self, node, name=None, id=None):
+        if name:
+            try:
+                name = (name or '').strip()
```

Comment:
I'm not really sure of the use of .strip() here, I can see why you would want to do this for the UX, but why remove whitespace before querying django?

Reply:
When you create a wiki, leading and trailing characters are removed from the page_name before it is saved.  If I am trying to retrieve that same wiki by looking up a page_name with white space, for example `" My wiki "`, and do `WikiPage.objects.get(page_name__icontains=" My wiki ", node=...,  deleted__isnull=True`), it will throw a DoesNotExist error.